<launch>
  <!--arg name="astra"   default="$(env USB_CAMERA)"  doc="Set astra(astra or usb camera) is true , set d435 is false"/> -->
  <include file="$(find astra_launch)/launch/astra.launch"/>
   <arg   name="depth_image"        default="/camera/depth/image_rect"/>
  <!--
  <arg     if="$(arg astra)" name="depth_image"        default="/camera/depth/image_rect"/>
  <arg unless="$(arg astra)" name="depth_image"      default="/camera/depth/image_rect_raw"/>
  -->
  <!--
  <group if="$(arg astra)">
     <include file="$(find astra_launch)/launch/astra.launch"/>
  </group>

  <group unless="$(arg astra)">
     <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" />
  </group>
  -->
  <node pkg="clbrobot_follower" type="clbrobot_follower_node" name="clbrobot_follower_node" output="screen">
    <remap from="depth_image" to="$(arg depth_image)" />
    <param name="enabled" value="true" />  <!-- enable/disable following -->
    <param name="camera_model" value="astra" />
     <!--
    <param name="camera_model" value="$(arg astra)" /> --> <!-- enable/disable following -->
    
    <param name="x_scale" value="2.5" />   <!-- turn amplitude facor -->
    <param name="z_scale" value="2.0" />   <!-- ignored -->
    <param name="min_x" value="-0.35" />  <!-- scan box size in meters, horiz from ctr -->
    <param name="max_x" value="0.35" />   <!-- scan box size in meters, horiz from ctr -->
    <param name="min_y" value="0.1" /> <!-- scan box size in meters, distance up from floor -->
    <param name="max_y" value="0.5" /> <!-- scan box size in meters, distance up from floor -->
    <param name="max_z" value="1.3" /> <!-- scan box size in meters, horiz depth from camera -->
    <param name="goal_z" value="0.7" /> <!-- distance to hold target centroid, distance from camera -->
    <param name="z_tol" value="0.1" />  <!-- +/- tolerance distance to hold target centroid, distance from camera -->
    <param name="x_tol" value="0.15" /> <!-- +/- tolerance distance to hold target centroid, lateral -->
  </node>
	
</launch>

